lib/Doctrine/Common/Lexer/AbstractLexer.php

```diff
@@ -18,6 +18,8 @@
 
 /**
  * Base class for writing simple lexers, i.e. for creating small DSLs.
+ *
+ * @psalm-type Token = array{value: int|string, type:string|int|null, position:int}
```

Comment:
I'm unsure if we should really allow `int` as a value, or if `dataProvider` should be fixed to quote the `10`